src/components/Patient/ManagePatients.tsx

```diff
@@ -804,169 +803,116 @@ export const PatientManager = () => {
-          <div className="mb-2 flex w-full flex-col items-center lg:mb-0 lg:w-fit lg:flex-row lg:gap-5">
-            <ButtonV2
-              id="add-patient-details"
-              onClick={() => {
-                const showAllFacilityUsers = ["DistrictAdmin", "StateAdmin"];
+    <>
+      <div className="flex w-full flex-col items-center justify-between lg:flex-row">
+        <div></div>
+        <div className="flex w-full flex-col items-center justify-end gap-2 lg:ml-3 lg:w-fit lg:flex-row lg:gap-3">
+          <Tabs
+            tabs={[
+              { text: t("live"), value: 0 },
+              { text: t("discharged"), value: 1 },
+            ]}
+            onTabChange={(tab) => {
+              if (tab === 0) {
+                updateQuery({ is_active: "True" });
+              } else {
+                const id = qParams.facility || onlyAccessibleFacility?.id;
+                if (id) {
+                  navigate(`facility/${id}/discharged-patients`);
+                  return;
+                }
+
                 if (
-                  qParams.facility &&
-                  showAllFacilityUsers.includes(authUser.user_type)
-                )
-                  navigate(`/facility/${qParams.facility}/patient`);
-                else if (
-                  qParams.facility &&
-                  !showAllFacilityUsers.includes(authUser.user_type) &&
-                  authUser.home_facility_object?.id !== qParams.facility
-                )
-                  Notification.Error({
-                    msg: "Oops! Non-Home facility users don't have permission to perform this action.",
-                  });
-                else if (
-                  !showAllFacilityUsers.includes(authUser.user_type) &&
-                  authUser.home_facility_object?.id
+                  authUser.user_type === "StateAdmin" ||
+                  authUser.user_type === "StateReadOnlyAdmin"
                 ) {
-                  navigate(
-                    `/facility/${authUser.home_facility_object.id}/patient`,
-                  );
-                } else if (onlyAccessibleFacility)
-                  navigate(`/facility/${onlyAccessibleFacility.id}/patient`);
-                else if (
-                  !showAllFacilityUsers.includes(authUser.user_type) &&
-                  !authUser.home_facility_object?.id
-                )
-                  Notification.Error({
-                    msg: "Oops! No home facility found",
-                  });
-                else setShowDialog("create");
-              }}
+                  updateQuery({ is_active: "False" });
+                  return;
+                }
+
+                Notification.Warn({
+                  msg: t("select_facility_for_discharged_patients_warning"),
+                });
+                setShowDialog("list-discharged");
+              }
+            }}
+            currentTab={tabValue}
```

Comment:
I really don't understand this login, if the user doesn't have access the API will return empty right 🤔

Reply:
I didn't write this, but it seems to be correct, or I did not understand you. It will prompt the user to select a facility if they are not a state admin. 🤔